src/sparc/client/services/metadata.py

```diff
@@ -123,103 +121,99 @@ def close(self) -> None:
+        
+        with requests.Session() as url_session:  
 
-        retries = Retry(
-            total=6, backoff_factor=1, status_forcelist=[403, 404, 413, 429, 500, 502, 503, 504]
-        )
+            retries = Retry(
+                total=6, backoff_factor=1, status_forcelist=[403, 404, 413, 429, 500, 502, 503, 504]
+            )
 
-        url_session.mount("https://", HTTPAdapter(max_retries=retries))
-
-        self.es_success = 1
-
-        try:
-            if headers == "NONE":
-                url_result = url_session.get(url)
-            else:
-                url_result = url_session.get(url, headers=headers)
-
-            if url_result.status_code == 410:
-                logging.warning("Retrieval Status 410 - URL Unpublished:" + url)
-            else:
-                url_result.raise_for_status()
-
-        except requests.exceptions.HTTPError as errh:
-            logging.error("Retrieving URL - HTTP Error:", errh)
-            self.es_success = 0
-        except requests.exceptions.ConnectionError as errc:
-            logging.error("Retrieving URL - Error Connecting:", errc)
-            self.es_success = 0
-        except requests.exceptions.Timeout as errt:
-            logging.error("Retrieving URL - Timeout Error:", errt)
-            self.es_success = 0
-        except requests.exceptions.RequestException as err:
-            logging.error("Retrieving URL - Something Else", err)
-            self.es_success = 0
-
-        url_session.close()
-
-        if self.es_success == 1:
-            result = url_result
-        else:
-            result = {}
+            url_session.mount("https://", HTTPAdapter(max_retries=retries))
+
+            self.es_success = 1
+
+            try:
+                if headers == "NONE":
+                    url_result = url_session.get(url)
+                else:
+                    url_result = url_session.get(url, headers=headers)
+
+                if url_result.status_code == 410:
+                    logging.warning("Retrieval Status 410 - URL Unpublished:" + url)
+                else:
+                    url_result.raise_for_status()
+
+            except requests.exceptions.HTTPError as errh:
+                logging.error("Retrieving URL - HTTP Error:", errh)
+                self.es_success = 0
+            except requests.exceptions.ConnectionError as errc:
+                logging.error("Retrieving URL - Error Connecting:", errc)
+                self.es_success = 0
+            except requests.exceptions.Timeout as errt:
+                logging.error("Retrieving URL - Timeout Error:", errt)
+                self.es_success = 0
+            except requests.exceptions.RequestException as err:
+                logging.error("Retrieving URL - Something Else", err)
+                self.es_success = 0
 
-        return result.json()
+            url_session.close()
+
+            result = url_result if self.es_success == 1 else {}
+
+            return result.json()
```

Comment:
`retrun url_result.json() if self.es_success == 1 else {}`
I can only assume that this doesn't have 100% line coverage because a python dict doesn't have an attribute `json`.

Reply:
Done